pushers/processor.go

```diff
@@ -44,34 +48,46 @@ type edLog struct {
 
 type edMetric struct {
 	common
-	DurationMs       float64 `json:"duration_ms"`
-	BilledDurationMs float64 `json:"billed_duration_ms"`
-	MaxMemoryUsed    float64 `json:"max_memory_used"`
-	MemorySize       float64 `json:"memory_size"`
+	BilledDurationMs      float64  `json:"billed_duration_ms"`
+	InitDurationMs        float64  `json:"init_duration_ms"`
+	RuntimeDurationMs     *float64 `json:"runtime_duration_ms,omitempty"`
+	DurationMs            float64  `json:"duration_ms"`
+	PostRuntimeDurationMs float64  `json:"post_runtime_duration_ms"`
+	MaxMemoryUsed         float64  `json:"max_memory_used"`
+	MemorySize            float64  `json:"memory_size"`
+	MemoryLeft            *float64 `json:"memory_left,omitempty"`
+	MemoryPercent         *float64 `json:"memory_percent,omitempty"`
+}
+
+type requestDurations struct {
+	Start time.Time
+	End   time.Time
 }
 
 type Processor struct {
-	tags         map[string]string
-	cloud        *cloud
-	outC         chan []byte
-	inC          chan []*lambda.LambdaEvent
-	invokeC      chan string
-	runtimeDoneC chan struct{}
-	stopC        chan struct{}
-	stoppedC     chan struct{}
+	tags             map[string]string
+	cloud            *cloud
+	outC             chan []byte
+	inC              chan []*lambda.LambdaEvent
+	invokeC          chan string
+	runtimeDoneC     chan struct{}
+	stopC            chan struct{}
+	stoppedC         chan struct{}
+	requestDurations *requestDurations
 }
 
 // NewProcessor initializes the log processor.
 func NewProcessor(conf *cfg.Config, outC chan []byte, inC chan []*lambda.LambdaEvent, runtimeDoneC chan struct{}) *Processor {
 	return &Processor{
-		outC:         outC,
-		inC:          inC,
-		invokeC:      make(chan string),
-		runtimeDoneC: runtimeDoneC,
-		stopC:        make(chan struct{}),
-		stoppedC:     make(chan struct{}),
-		tags:         conf.Tags,
-		cloud:        &cloud{ResourceID: conf.FunctionARN},
+		outC:             outC,
+		inC:              inC,
+		invokeC:          make(chan string),
+		runtimeDoneC:     runtimeDoneC,
+		stopC:            make(chan struct{}),
+		stoppedC:         make(chan struct{}),
+		tags:             conf.Tags,
+		cloud:            &cloud{ResourceID: conf.FunctionARN, AccountID: conf.AccountID, Region: conf.Region},
+		requestDurations: &requestDurations{},
```

Comment:
we should this per request id, to make sure that end corresponds to the right start ( start might come before the previous request ends)